velox/exec/tests/utils/PlanBuilder.h

```diff
@@ -1114,23 +1114,43 @@ class PlanBuilder {
       const std::vector<std::string>& outputLayout,
       core::JoinType joinType = core::JoinType::kInner);
 
+  /// Presents a index join condition.
+  struct IndexJoinCondition {
+    /// The type of the join condition, and we currently support "in" and
+    /// "between".
+    std::string type;
+    /// The name of index columns from right side.
+    std::string indexColumn;
+    /// The list of condition columns or constants from left side.
+    std::vector<std::string> conditionInputs;
+  };
+
+  static IndexJoinCondition betweenIndexJoinCondition(
+      const std::string& indexColumn,
+      const std::string& lowerBound,
+      const std::string& upperBound);
+
+  static IndexJoinCondition inIndexJoinCondition(
+      const std::string& indexColumn,
+      const std::string& listColumn);
+
   /// Add an IndexLoopJoinNode to join two inputs using one or more join keys
   /// plus optional join conditions. First input comes from the preceding plan
   /// node. Second input is specified in 'right' parameter and must be a
   /// table source with the connector table handle with index lookup support.
   ///
   /// @param right The right input source with index lookup support.
-  /// @param joinCondition SQL expressions as the join conditions. Each join
-  /// condition must use columns from both sides. For the right side, it can
-  /// only use one index column.
+  /// @param joinCondition A list of join conditions defined by
+  /// 'IndexJoinCondition'. Each join condition must use columns from both
+  /// sides. For the right side, it can only use one index column.
   /// @param joinType Type of the join supported: inner, left.
   ///
   /// See hashJoin method for the description of the other parameters.
   PlanBuilder& indexLookupJoin(
       const std::vector<std::string>& leftKeys,
       const std::vector<std::string>& rightKeys,
       const core::TableScanNodePtr& right,
-      const std::vector<std::string>& joinCondition,
+      const std::vector<IndexJoinCondition>& joinConditions,
```

Comment:
Can we keep this as a SQL string as it is easier to use? Let's document what SQL strings are supported.

Reply:
Good poiint.